codegen/tests/integration/custom_tracing_layer.rs

```diff
@@ -0,0 +1,59 @@
+use reqwest::Client;
+use serde_json::Value as JsonValue;
+use shuttle_common_tests::cargo_shuttle::cargo_shuttle_run;
+use tokio::time::Duration;
+
+#[tokio::test]
+async fn custom_tracing_layer() {
+    // Spin up the example
+    let path = "../examples/tracing/axum-logs-endpoint";
+    let url = cargo_shuttle_run(path, false).await;
+
+    // Prepare URLs
+    let get_url = format!("{url}/logs/3");
+    let post_url1 = format!("{url}/message/hello");
+    let post_url2 = format!("{url}/message/world");
+    let post_url3 = format!("{url}/message/how%20are%20you%3F");
+
+    let client1 = Client::new();
+    let client2 = client1.clone();
+
+    // Send the initial GET request
+    let get = tokio::spawn(async move {
+        client1
+            .get(get_url)
+            .send()
+            .await
+            .unwrap()
+            .json::<Vec<JsonValue>>()
+            .await
+            .unwrap()
+    });
+
+    // Wait for the request to send
+    tokio::time::sleep(Duration::from_millis(500)).await;
```

Comment:
Yeah, this test can take a more sequential/synchronous approach.

Reply:
The way the axum server works is that you make a GET request and it will wait for enough messages to come in before it responds, so you need to make the GET request first. The 500ms could probably be lower, I just chose a random value that worked consistently.

I couldn't think of a better way to make this work because it is, as already established, not very practical, and I just didn't come up with anything better than that 😅

Reply:
Alright, got a bit confused.